dev-lang/rust/rust-1.27.0.recipe

```diff
@@ -0,0 +1,236 @@
+	ln -s $sourceDir15/stdsimd-$srcGitRev15 src/stdsimd
+	ln -s $sourceDir16/rust-clippy-$srcGitRev16 src/tools/clippy
+
+	# write the build configuration
+	tr -d '\t' >config.toml <<- EOL
+	[llvm]
+	targets = "X86"
+	experimental-targets = ""
+
+	[build]
+	cargo = "/$relativeBinDir/cargo"
+	rustc = "/boot/system/bin/rustc"
+	submodules = false
+	extended = true
+	tools = ["cargo", "rls", "rustfmt", "analysis"]
+
+	[install]
+	prefix = "$prefix"
+	libdir = "$relativeLibDir"
+	mandir = "$relativeManDir"
+	docdir = "$relativeDevelopDocDir"
+	sysconfdir = "$relativeDataDir"
+
+	[rust]
+	channel = "stable"
+	use-jemalloc = false
+	rpath = false
+	deny-warnings = false
+	dist-src = false
+
+	[dist]
+	src-tarball = false
+EOL
+	# Disable ASLR: compiling stage 1 rustc requires a lot of RAM (about 1.5
+	# GB). Haiku has a per-process limit of 2GB on 32 bit systems. ASLR makes
+	# the available space even smaller. Disabling it will give us the space to
+	# compile Rust
+	export DISABLE_ASLR=1
+
+	# now build rust and cargo
+	./x.py dist
+}
+
+INSTALL()
+{
+	# we will manually invoke the install scripts
+	if [ $effectiveTargetArchitecture = x86 ]; then
+		architecture="i686-unknown-haiku"
+	fi
+	if [ $effectiveTargetArchitecture = x86_64 ]; then
+		architecture="x86_64-unknown-haiku"
+	fi
+
+	# let's install the packages one by one
+	cd $sourceDir/build/tmp/dist/
+	for module in "rust-docs-$srcGitRev-$architecture"     \
+	              "rust-std-$srcGitRev-$architecture"      \
+	              "rustc-$srcGitRev-$architecture"         \
+	              "rust-analysis-$srcGitRev-$architecture" \
+	              "cargo-$cargoVersion-$architecture"      \
+	              "rls-$rlsVersion-$architecture"          \
+	              "rustfmt-$rustfmtVersion-$architecture"
+	do
+		./$module/install.sh                               \
+				--prefix=$prefix                           \
+				--docdir=$developDocDir                    \
+				--libdir=$libDir                           \
+				--mandir=$manDir                           \
+				--sysconfdir=$dataDir                      \
+				--disable-ldconfig
+	done
+
+	# move the cargo and binaries (in case of a secondary arch)
+	if [ -n "$secondaryArchSuffix" ]; then
+		mkdir -p $binDir
+		mv $prefix/bin/cargo $binDir/cargo
+	fi
+
+	# remove zsh data, it is not used on Haiku anyway
+	rm -rf $prefix/share
+
+	# move the `rustlib` folder to the developLibDirs (as it is a framework of sorts)
+	# do create a link in $prefix/lib as that is where rustc expects things to live
+	# Note; this actually seems to be a bug in the Rust build system. The path
+	# to rustlib is hardcoded in the rustc binary, but it does allow it to be
+	# set to libdir_relative (see config.rs in the bootstrap tool). This variable
+	# is only set when the configure script is used to generate the config, not
+	# with config.toml
+	mkdir -p $developLibDir
+	mv $libDir/rustlib $developLibDir
+	cd $prefix/lib
+	ln -s $developLibDir/rustlib rustlib
```

Comment:
Suggestion: make the symlink relative.
```diff
-	cd $prefix/lib
-	ln -s $developLibDir/rustlib rustlib
+	ln -r -s $developLibDir/rustlib $prefix/lib/rustlib
```

BTW, it would be cool if we could teach rust to look for the  `rustlib` subdirectory/symlink in `$libDir` instead of `$prefix/lib`, because, on secondary arch builds, the directory in which we're creating this symlink is not really the right one ;)
(But I guess we can keep this as is for now.)